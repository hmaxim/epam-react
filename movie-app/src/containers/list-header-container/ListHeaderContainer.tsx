import React, { useCallback, useState, useEffect, useMemo } from 'react';
import ListHeaderWrapper from './components/ListHeaderWrapper';
import ListHeaderTitle from './components/ListHeaderTitle';
import ListHeaderSwitchersContainer from './components/ListHeaderSwitchersContainer';
import SwitchersButtons from '../../shared-components/switchers-buttons/SwitchersButtons';
import { setSearchParams } from '../../redux/rootActions';
import { connect } from 'react-redux';
import { useQuery, getNavUrl } from '../../constants/constants';
import { useHistory, useLocation } from 'react-router';

const ListHeaderContainer = (props: any) => {
  const history = useHistory();
  const location = useLocation();
  const query = useQuery(location.search);
  const isActive = useCallback(
    (val: string): boolean => val === query.get('sortBy'),
    [],
  );

  const sortButtons = useMemo(
    () => [
      {
        label: 'Release Date',
        active: isActive('release_date') || true,
        buttonValue: 'release_date',
      },
      {
        label: 'Rating',
        active: isActive('vote_count'),
        buttonValue: 'vote_count',
      },
    ],
    [],
  );

  const navigate = useCallback(() => {
    history.push({
      pathname: '/search',
      search: getNavUrl(props.searchParams),
    });
  }, [props.searchParams]);

  useEffect(() => {
    if (
      query.get('sortBy') &&
      props.searchParams.sortBy !== query.get('sortBy') &&
      !history.location.pathname.includes('film')
    ) {
      navigate();
    }
  }, [props.searchParams.sortBy]);

  const callback = useCallback((buttonValue: string) => {
    props.setSearchParams({ sortBy: buttonValue });
    if (!query.get('sortBy')) {
      navigate();
    }
  }, []);

  return (
    <ListHeaderWrapper>
      <ListHeaderTitle>{props.listHeaderTitle}</ListHeaderTitle>
      <ListHeaderSwitchersContainer>
        {history.location.pathname.includes('search') ? (
          <SwitchersButtons
            switchersTitle={'SORT BY'}
            click={callback}
            activeBtnValue={props.searchParams.sortBy}
            searchButtonsParams={sortButtons}
          ></SwitchersButtons>
        ) : null}
      </ListHeaderSwitchersContainer>
    </ListHeaderWrapper>
  );
};

const mapStateToProps = (state: any) => {
  return {
    searchParams: state.searchParams,
  };
};
const mapDispatchToProps = {
  setSearchParams,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ListHeaderContainer);
