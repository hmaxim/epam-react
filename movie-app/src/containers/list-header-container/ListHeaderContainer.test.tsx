import 'jsdom-global/register';
import React from 'react';
import { render } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import ListHeaderContainer from './ListHeaderContainer';
import toJson from 'enzyme-to-json';

describe('ListHeaderContainer component', () => {
  let component: any;

  beforeEach(() => {
    component = render(<ListHeaderContainer />);
  });

  test('should be rendered', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
