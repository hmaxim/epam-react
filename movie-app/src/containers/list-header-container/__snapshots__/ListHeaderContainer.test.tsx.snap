// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeaderContainer component should be rendered 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 9.5%;
  color: white;
  background: #555555;
}

.c0 .search-buttons-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  padding: 20px 0;
}

.c0 .search-buttons-container span {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-right: 30px;
  font-size: 20px;
}

.c0 .search-buttons-container button {
  width: auto;
  border: none;
  cursor: pointer;
  height: 30px;
}

.c1 {
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-left: 65%;
}

.c3 {
  background: #F65261;
  border-radius: 3px;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 0px;
  border-top-right-radius: 0px;
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
}

.c4 {
  background: rgba(85,85,85,0.5);
  border-radius: 3px;
  border-top-left-radius: 0px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 3px;
  border-top-right-radius: 3px;
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      class="search-buttons-container"
    >
      <span>
        SORT BY
      </span>
      <button
        class="c3"
        id="Release Date"
      >
        Release Date
      </button>
      <button
        class="c4"
        id="Rating"
      >
        Rating
      </button>
    </div>
  </div>
</div>
`;
